kong/plugins/rate-limiting/handler.lua

```diff
@@ -28,6 +28,13 @@ local function get_identifier(conf)
     end
   elseif conf.limit_by == "credential" then
     identifier = ngx.ctx.authenticated_credential and ngx.ctx.authenticated_credential.id
+  elseif conf.limit_by == "http_header" then
+    local header = ngx.req.get_headers()[conf.http_header]
```

Comment:
im not sure calling `get_headers`, which returns a table, 3 times is the best solution. imo it would be better to localize this via the `ngx.var` table (e.g. `local header_name = conf.http_header; local header = ngx.var.http_ .. "header_name"`) and the examine as such.

also, im not really sure about the logic to simply look at the last header if multiples are present. we dont do this anywhere else. to me it makes more sense to reject the request with a 400, as it's clearly malformed in this case.